### PR TITLE
added cell background of empty rows

### LINKED ISSUE #2868: [Bug] Background colour does not get added into the entire table
Steps to reproduce:
1) Add a table widget
2) Click on the property pane of the table 
3) Navigate to Background colour
4) Add a colour 
5) Ensure the colour is added 
6) Now scroll down 
and observe

Observation:
It is observed that the colour is not added to the entire table ie. The BG colour is added to only those rows that have data. If the data is not available then the row is displayed as white 

Expectation:
It must be added to the entire table irrespective of the data being added or not 

Screenshot 2021-02-04 at 11 53 04 AM

### PR DESCRIPTION
**Pull request by haojin111**
## Description

> When apply cell background of table property, it should be applied to empty row's cells.

Fixes #2868 

> if no issue exists, please create an issue and ask the maintainers about this first

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please refer attached issue

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:red_circle: Total coverage has decreased</summary>

    // Code coverage diff between base branch:release and head branch: FIX/2868-issue-of-adding-background-table 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :red_circle: | total | 51.7 **(0)** | 33.62 **(-0.02)** | 29.55 **(0)** | 52.35 **(-0.01)**
 :red_circle: | app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx | 38.02 **(-0.96)** | 16.85 **(-2.14)** | 9.38 **(0)** | 36.04 **(-1)**</details>

### CODE REVIEW

File: app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx [hunk 1]
```diff
@@ -284,9 +284,16 @@ export const renderEmptyRows = (
...
-              <div {...cell.getCellProps()} className="td" key={cellIndex} />
-            );
+            const cellProps = cell.getCellProps();
+            const background =
+              columns[0]?.columnProperties?.cellBackground || null;
+            if (background) {
+              cellProps.style = {
+                ...cellProps.style,
+                background,
+              };
```
Comment (SatishGandham) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx:294]: change this to `cellProps.style.background:background`, creating a new style object will cause extra renders.

**Review by SatishGandham — CHANGES_REQUESTED** [review]

File: app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx [hunk 2]
```diff
@@ -284,9 +284,10 @@ export const renderEmptyRows = (
         <div {...rowProps} className="tr" key={index}>
           {multiRowSelection && renderCheckBoxCell(false)}
           {row.cells.map((cell: any, cellIndex: number) => {
-            return (
-              <div {...cell.getCellProps()} className="td" key={cellIndex} />
-            );
+            const cellProps = cell.getCellProps();
+            cellProps.style.background =
+              columns[0]?.columnProperties?.cellBackground || null;
+            return <div {...cellProps} className="td" key={cellIndex} />;
```
Comment (SatishGandham) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx:290]: Can you write it like this. saves extra call to getCellProps.
`if(columns[0]?.columnProperties?.cellBackground){
const cellProps = cell.getCellProps();
            cellProps.style.background =
              columns[0]?.columnProperties?.cellBackground
}`

**Review by SatishGandham — CHANGES_REQUESTED** [review]

File: app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx (hunk 2 above)
Comment (haojin111) [reply app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx:290]: `const cellProps = cell.getCellProps();
 if (columns[0]?.columnProperties?.cellBackground) {
    cellProps.style.background =
      columns[0].columnProperties.cellBackground;
 }
 return <div {...cellProps} className="td" key={cellIndex} />;`
How about this?

Comment (SatishGandham) [reply app/client/src/components/designSystems/appsmith/TableComponent/TableUtilities.tsx:290]: Works. I didn't notice cellprops was used below.

**Review by SatishGandham — APPROVED** [review]

**Review by riodeuno — APPROVED** [review]

**Comment by riodeuno** [comment]
@somangshu This was reprioritised to a lower priority deferring to other issues, which were more critical at the time.
@haojin111 Thanks a lot for picking this up! The code looks fine to me.  Tested this update on the deploy preview as well.

**Pull request merged**